ffs/filcold/filcold.go

```diff
@@ -326,53 +383,6 @@ Loop:
 	return ffs.FilStorage{}, fmt.Errorf("aborted due to cancellation")
 }
 
-// estimatePieceSize estimates the size of the Piece that will be built by the filecoin client
-// when making the deal. This calculation should consider the unique node sizes of the DAG, and
-// padding. It's important to not underestimate the size since that would lead to deal rejection
-// since the miner won't accept the further calculated PricePerEpoch.
-func (fc *FilCold) calculatePieceSize(ctx context.Context, c cid.Cid) (uint64, error) {
-	// Get unique nodes.
-	seen := cid.NewSet()
-	if err := dag.Walk(ctx, fc.getLinks, c, seen.Visit); err != nil {
-		return 0, fmt.Errorf("walking dag for size calculation: %s", err)
-	}
-
-	// Account for CAR header size.
-	carHeader := car.CarHeader{
-		Roots:   []cid.Cid{c},
-		Version: 1,
-	}
-	totalSize, err := car.HeaderSize(&carHeader)
-	if err != nil {
-		return 0, fmt.Errorf("calculating car header size: %s", err)
-	}
-
-	// Calculate total unique node sizes.
-	buf := make([]byte, 8)
-	f := func(c cid.Cid) error {
-		s, err := fc.ipfs.Block().Stat(ctx, path.IpfsPath(c))
-		if err != nil {
-			return fmt.Errorf("getting stats from DAG node: %s", err)
-		}
-		size := uint64(s.Size())
-		carBlockHeaderSize := uint64(binary.PutUvarint(buf, size))
-		totalSize += carBlockHeaderSize + size
-		return nil
-	}
-	if err := seen.ForEach(f); err != nil {
-		return 0, fmt.Errorf("aggregating unique nodes size: %s", err)
-	}
-
-	// Consider padding.
-	paddedSize := padreader.PaddedSize(totalSize).Padded()
-
-	return uint64(paddedSize), nil
-}
-
-func (fc *FilCold) getLinks(ctx context.Context, c cid.Cid) ([]*format.Link, error) {
-	return fc.ipfs.Object().Links(ctx, path.IpfsPath(c))
-}
-
```

Comment:
I was proud of doing this code since there wasn't a good way to calculate the right piece size... but has to go now.

Reply:
😿